doc/source/ray-observability/user-guides/ray-profiling.rst

```diff
@@ -0,0 +1,83 @@
+.. _ray-profiling:
+
+Profiling on Ray
+================
+Profiling is one of the most important debugging tools to diagnose performance, out of memory, hanging, or other application issues as desribed `here <https://docs.ray.io/en/master/ray-observability/user-guides/profiling.html>`_. For better experience, Ray is natively supporting profilers to be run for task, actor or job in distributed cases.
+
+
+.. note::
+
+    Ray Profiling is under active development with more profilers to be natively supported in the future. APIs are subject to change.
+
+GPU Profiling
+--------------
+
+Installation
+~~~~~~~~~~~~
+
+First, install the Nsight System CLI by following the `Nsight User Guide <https://docs.nvidia.com/nsight-systems/InstallationGuide/index.html>`_.
+
+Confirm if Nsight System is installed by running:
+
+.. code-block:: bash
+
+  $ nsys --version
+
+  # NVIDIA Nsight Systems version 2022.4.1.21-0db2c85
+
+Run Nsight on Ray
+~~~~~~~~~~~~~~~~~~~
+
+To enable GPU profiling, you can specify the config in `runtime_env` as follows:
+
+.. testcode::
+
+  import ray
+
+  ray.init()
+
+  @ray.remote(runtime_env={ "nsight": "default" })
+  def ray_task():
+    ...
+  
+  # run nsight with config: "nsys profile [default options] ..."
+  ray.get(ray_task.remote())
+
+The `"default"` config can be found in `nsight.py <https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/nsight.py#L20>`_.
+
+Custom Options
+**************
+
+You can also add custom options for nsight profiler by specfying a dictionary of option values.
```

Comment:
1. I would link the options: https://docs.nvidia.com/nsight-systems/UserGuide/index.html#cli-launch-command-switch-options
2. Capitalize "Nsight"